book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,266 @@
+and pickles. Given that they are implemented in different languages and exist in
+different repositories special considerations each layer will be required.
+
+## Where to run the benchmarks
+
+To start everyone with the resources should be able to run the benchmarks
+locally, additionally we may want some of the benchmarks to run in CI to detect
+regressions. Some benchmarks may be considerably expensive to run, a decision
+will have to be made about which benchmarks to run and how often we should run
+them.
+
+## Parameters
+
+It will be of interest parameterizing some benchmarks, for example benchmarking
+verification time but doing it with different circuits sizes, that would show
+how performance scales with the size of the circuit.  The introduction of
+parameters doesn't cause any fundamental change to the benchmarks, but it can
+considerably increase the cost of running them, making them more the kind of
+benchmarks that we may not want to run every single time in CI.
+
+## Benchmark classes
+
+Next the classes of benchmarks to consider, they are different combinations of
+metrics, layers and consumers.
+
+| benchmarks               | repository    | metric   | for external use | for internal use | regression test | noise sensitive  |
+| ------------------------ | ------------- | -------- | ---------------- | ---------------- | --------------- | ---------------- |
+| kimchi time              | proof-systems | time     | yes              | yes              | yes             | yes              |
+| kimchi time flamegraph   | proof-systems | time     | no               | yes              | no              | no               |
+| kimchi instructions      | proof-systems | time     | no               | yes              | yes             | no               |
+| kimchi memory            | proof-systems | memory   | yes              | yes              | yes             | no               |
+| kimchi memory flamegraph | proof-systems | memory   | no               | yes              | no              | no               |
+| snarky time              | mina/snarky*  | time     | yes              | yes              | yes             | yes              |
+| snarky memory            | mina/snarky*  | memory   | yes              | yes              | yes             | no               |
+| pickles time             | mina/pickles* | time     | yes              | yes              | yes             | yes              |
+| pickles memory           | mina/pickles* | memory   | yes              | yes              | yes             | no               |
+
+\* these have to be in Mina now because pickles is in Mina, and while snarky has
+its own repository, using it requires a few things still in Mina, so that we
+might want to consider some changes to make these benchmarks easier and probably
+cheaper.
+
+### Kimchi time
+
+One of the simplest measuring running time, can be a specific case or allow more
```

Comment:
Which will we choose? A specific case? What is the case? Something more flexible? Can you describe it in detail?

Reply:
With specific case I mean without parameters. I think it would be a per benchmark decision, we could choose for example circuit size $2^{16}$ that is similar to the mina circuits or have more flexibility through parameters and benchmark a range of sizes.